extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/QuarkusPathLocationScanner.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.flyway.runtime;
+
+import java.io.IOException;
+import java.net.JarURLConnection;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import org.flywaydb.core.api.Location;
+import org.flywaydb.core.api.logging.Log;
+import org.flywaydb.core.api.logging.LogFactory;
+import org.flywaydb.core.internal.resource.LoadableResource;
+import org.flywaydb.core.internal.resource.classpath.ClassPathResource;
+import org.flywaydb.core.internal.scanner.classpath.ResourceAndClassScanner;
+
+public final class QuarkusPathLocationScanner implements ResourceAndClassScanner {
+    private static final Log LOG = LogFactory.getLog(QuarkusPathLocationScanner.class);
+    // TODO: this should be configurable, using flyway default
+    private final static String DEFAULT_NATIVE_LOCATION = "db/migration";
+    private static Set<String> ALL_SQL_RESOURCES;
+
+    static {
+        try {
+            ALL_SQL_RESOURCES = discoverApplicationMigrations();
```

Comment:
nice, I like this solution.

Incidentally, I think it's totally ok to not support reconfiguration of `DEFAULT_NATIVE_LOCATION`. Having less configuration properties makes for better usability.